src/groups.py

```diff
@@ -2,27 +2,27 @@
 # all dependancy on nodeattr is to be located in this file
 import config
 
+import click
 from plumbum import local, ProcessExecutionError
 from tempfile import NamedTemporaryFile
-from click import ClickException
 from re import search
 
 def list_groups():
     groups = __nodeattr(arguments=['-l'])
-    groups.remove('')
+    while '' in groups: groups.remove('')
     return groups
 
 def nodes_in(group_name):
     nodes = __nodeattr(arguments=['-n', group_name])
-    nodes.remove('')
+    while '' in nodes: nodes.remove('')
     return nodes
 
 def expand_nodes(node_list):
     # intercept to generate a more useful error message
     #   before invalid nodenames are caught generically in __nodeattr
     for node in node_list:
         if search(r'[^A-z0-9,\[\]]', node):
-            raise ClickException("""
+            raise click.ClickException("""
 Invalid nodename {} - may only contain alphanumerics, ',', '[' and ']'
```

Comment:
I was going to say the "and" is missing the oxford common, however the comment is essentially line noise already. Instead the comment should be updated to.
```
Invalid nodename {} - may only contain alphanumerics characters and the following symbols: , [ ]
```

Reply:
https://github.com/alces-software/adminware/pull/169/commits/fa3fa3f33723bea056474b2db82e7b49e60f548b